clients/src/main/java/org/apache/kafka/common/errors/InvalidShareSessionEpochException.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.common.errors;
+
+public class InvalidShareSessionEpochException extends RetriableException {
```

Comment:
@AndrewJSchofield Though I do see `INVALID_SHARE_SESSION_EPOCH` mentioned in the KIP but do we have to mention this new exception class as well in the KIP?

Reply:
Probably should. I'll make a note to add it.

Reply:
I think same for the other exception class in the PR as well then, there seems to be 2 new classes.